File: backend/ai-assistant-service/src/main/java/com/bbmovie/ai_assistant_service/core/low_level/_controller/TestController.java
package com.bbmovie.ai_assistant_service.core.low_level._controller;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.oauth2.jwt.Jwt;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/v1/test")
public class TestController {

    @GetMapping("/public")
    public Mono<String> publicEndpoint() {
        return Mono.just("Public access works!");
    }

    @GetMapping("/authenticated")
    public Mono<String> authenticatedEndpoint(@AuthenticationPrincipal Jwt jwt) {
        return Mono.just("Authenticated as: " + jwt.getSubject());
    }

    @GetMapping("/optional")
    public Mono<String> optionalAuth(
            @AuthenticationPrincipal(errorOnInvalidType = false) Jwt jwt) {
        return Mono.just(jwt != null 
            ? "Authenticated as: " + jwt.getSubject()
            : "Anonymous access");
    }
}

File: backend/watchlist/src/main/kotlin/com/bbmovie/watchlist/controller/TestController.kt
package com.bbmovie.watchlist.controller

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/test")
class TestController {

    @RequestMapping("/hello")
    fun test(): ResponseEntity<String> {
        return  ResponseEntity.status(HttpStatus.OK).body("Hello");
    }

    @PreAuthorize("isAuthenticated()")
    @RequestMapping("/secure")
    fun secureTest(): ResponseEntity<String> {
        return  ResponseEntity.status(HttpStatus.OK).body("Secure Hello");
    }
}